### PR TITLE
Reagent soup / Soup rework / Stoves - A kitchen expansion

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

image

**This PR:** 

- Reworks most* existing soup into reagents. 

- Adds Stoves and Ranges. Ranges replace most* existing ovens. 

- Adds soup pots, to cook soup

**How does it work?** 

In the kitchen you will find a stove now.

Stoves act as a "reagent container heater", essentially a chem heater. You can set a pot onto the stove. 

To make soup, visit the cooking recipe book for a guide. Most recipes are the same as before, just tweaked slightly - Add water to the pot (50 units for 1 batch generally), then add all the corresponding ingredients to the pot. Set the pot out on the stove and right click it to turn it on. If the recipe's correct, shortly it will start to mix and give you soup!

One soup recipe will give you roughly 3 servings of soup. You can pour our the soup into a bowl using a ladle or just by pouring it manually. 

Of note: **All of the reagent contents of the ingredient are transferred into the soup.** Better, more nutrient rich ingredients produces more soup, and poisoned produce will pass it on. 

If you place the soup into a chem master, you will notice it's roughly half "soup reagent" and half a variety of reagents, including nutriments / proteins. This is your soup! It is recommended you serve your soup with the reagents included, as they make up more nutrition for the customer, however you can separate it out if you're picky. 

**Todo:** 

- [x] Fill out the PR body a bit more 
- [x] Mapping (wait for big merge conflict pr to go past)
- [x] Soup colors
- [x] Balance pass over for soup recipes
- [x] TODOs
- [ ] Unit tests
- [x] Cullen Skink's recipe is invalid
- [x] Try to see if there's an easy way to prevent soup from fattening you up too easy. 

## Why it's good for the game

Adds some more depth to the kitchen and moves chef away from the click-button-get-food style that exists. 

Allows for inherently custom soups by the way of making it reagents, so no need to support custom soup food items. 

## Changelog

:cl: Melbert, stove and pot sprites by Kryson, ladle sprite by Kinneb
add: Kitchens are now stocked with Ranges. 
add: You can now print (and create) Stoves. 
add: The dinnerware vendor now dispenses ladles. 
add: Spoons can now actually spoon... things.
add: Soup has been reworked entirely. Soups are now reagents, cooked via a soup pot on a Stove or Range. Simply add water and your required items, then apply heat. Be careful not to boil over!
add: Stoves, Ranges, and Griddles will now heat up their surroundings - don't turn them on around plasma!
fix: Fixes being able to cook in an Oven while the room is depowered
qol: Hitting a customer bot with an incorrect recipe no longer counts as a hostile attack leading to your demise shortly after
refactor: Customer bots that request a reagent now use custom orders
code: Cut down a lot of code in the crafting menu code, and removes some ugly ispaths
del: Soup is no longer food items, so can't appear in random food pools (at least not yet). 
balance: Virus Food recipe now requires you cool it to 200k.
/:cl:

### CODE REVIEW

**Comment by DATA-xPUNGED** [comment]
Boiled rice should absolutely be made on these stoves

**Comment by CMDR-Gungnir** [comment]
> what do you mean you’re at soup

WHAT STORE ARE YOU IN

**Comment by Maurukas** [comment]
Melbert, how is it possible for you to be this awesome?

**Comment by Vect0r2** [comment]
I cannot wait

**Comment by dwasint** [comment]
this is a certified banger pr and reason enough for me to rebase to tg. Way better than the custom soup code i had which used a fake food item lol

**Comment by dwasint** [comment]
also this is just a wish from me but i'd be dope if you could add a looping sound for the water boiling in the pots as the pot will most likely be used for far more in the future

**Review by Imaginos16 — APPROVED** [review]
https://youtu.be/P3ALwKeSEYs

**Comment by Edelor68** [comment]
Does the stove only heat up pots or can you put other stuff on it as well?

**Comment by SomeoneYouProbablyKnow** [comment]
Finally. Cook meth cooking mechanics

**Comment by ModDownloading** [comment]
If you end up making stuff like spaghetti and rice use the cooking pot instead of the microwave, it would be really nice to give permabrig an oven too. Rice-based meals are some of the few things you can use perma's already tiny kitchen to cook and I'd still like that area to have some use (for a different PR, this one's great as it is).

**Comment by ModDownloading** [comment]
Possibly you could use wish soup as a base? It's literally just water, so if soup has no other reagents except water in the pot, it's "wish soup", but if there's something in the pot that doesn't match any recipe, including wish soup, it makes a "custom soup" using wish soup as the base. Examining it or any bowl of soup could show you all the solid ingredients that were added (but not the liquid ones, so it doesn't give away there's initropidril in there or something).

**Comment by Wallemations** [comment]
I'm going to make a soup store that exclusively sells clothes

File: code/modules/asset_cache/assets/crafting.dm [hunk 1]
```diff
@@ -2,33 +2,48 @@
...
+			icon_state = style.icon_state
+
+	icon_file ||= initial(preview_item.icon_preview) || initial(preview_item.icon)
+	icon_state ||= initial(preview_item.icon_state_preview) || initial(preview_item.icon_state)
+
+#ifdef UNIT_TESTS
+	if(!icon_exists(icon_file, icon_state, scream = TRUE))
 		return
-	#endif
+#endif
```
Comment (san7890) [inline code/modules/asset_cache/assets/crafting.dm:45]: Couldn't you change this to be a focus-only unit test with `PERFORM_ALL_TESTS`? Would be a lot more specific/self-documenting that this is a "testing" check, and not some assertion we pass in here for when we're running UNIT_TESTS.

**Review by san7890 — CHANGES_REQUESTED** [review]
would an updatepaths not work here?

File: code/modules/asset_cache/assets/crafting.dm (hunk 1 above)
Comment (san7890) [reply code/modules/asset_cache/assets/crafting.dm:45]: feels like a good idea to change it over imho

**Comment by Wallemations** [comment]
Since you added stovetops, how capable do you think the game would be at having a pan item that acts as a 1-2 item griddle when placed on a stove?

**Comment by dwasint** [comment]
> > Also, is there support for custom soup
> 
> Yes and no. All soup can now intrinsically be customized, since it's just reagents. You can pour whatever in.
> 
> You can also add any ingredients to a recipe that's mixing and it'll add its reagents into it, including deep fried stuff.
> 
> However it requires that any existing recipe is being made. So you can't just throw in ANYTHING and it will eventually mix. I do want to support that, but I can't think of a way to do it intuitively.

could modify reagent code to have a reaction tag that requires their to be a maximum number of reagents present for the reaction to occur than cap the soups at like 3-4 above their limits

File: code/__DEFINES/food.dm [hunk 2]
```diff
@@ -177,6 +177,14 @@ DEFINE_BITFIELD(food_flags, list(
 
 #define BLACKBOX_LOG_FOOD_MADE(food) SSblackbox.record_feedback("tally", "food_made", 1, food)
 
+/// Point water boils at
+#define WATER_BOILING_POINT 373
```
Comment (san7890) [inline code/__DEFINES/food.dm:181]: Make this something like

```suggestion
#define WATER_BOILING_POINT T0C + 100
```

Where T0C is 0 degrees celsius (273.15 Kelvin), and we can derive that number from there. Much more sciency!

File: code/__DEFINES/food.dm [hunk 3]
```diff
@@ -177,6 +177,14 @@ DEFINE_BITFIELD(food_flags, list(
...
+/// Point at which soups begin to burn at
+#define SOUP_BURN_TEMP 540
```
Comment (san7890) [inline code/__DEFINES/food.dm:183]: this should also be reliant on `WATER_BOILING_POINT` somehow

File: code/__HELPERS/randoms.dm [hunk 4]
```diff
@@ -4,28 +4,28 @@
...
-		/obj/item/food/meat/slab,
-		/obj/item/food/soup,
-		/obj/item/food/grown,
-		/obj/item/food/grown/mushroom,
-		/obj/item/food/clothing,
-		/obj/item/food/meat/slab/human/mutant,
-		/obj/item/food/grown/ash_flora,
-		/obj/item/food/grown/nettle,
-		/obj/item/food/grown/shell
+			/obj/item/food/drug,
```
Comment (san7890) [inline code/__HELPERS/randoms.dm:7]: alphabetize while you're in the area

File: code/modules/food_and_drinks/recipes/soup_mixtures.dm [hunk 5]
```diff
@@ -0,0 +1,1720 @@
...
+		/datum/reagent/consumable/nutriment/soup/clownchili = 30,
+		/datum/reagent/consumable/tomatojuice = 8,
+		/datum/reagent/consumable/laughter = 4,
+		/datum/reagent/consumable/banana = 4,
+	)
+	percentage_of_nutriment_converted = 0.15
+
+// Vegan Chili
+/datum/reagent/consumable/nutriment/soup/chili_con_carne
+	name = "Chili Sin Carne"
```
Comment (san7890) [inline code/modules/food_and_drinks/recipes/soup_mixtures.dm:495]: man this type path sucks for what it really is "chili con carne" path for "chili _sin_ carne"? come on. it made me think you somehow screwed up the "chili con carnival" but that has a `/clownchili` typepath so whatever

no need to change it but i wnated to mentoin it now before i forget

**Review by san7890 — CHANGES_REQUESTED** [review]
should be okay after the first three are addressed in some way

File: code/__DEFINES/food.dm (hunk 3 above)
Comment (san7890) [reply code/__DEFINES/food.dm:183]: i still think it would be neat if we could incorporate boiling -> burning but whatever

**Pull request merged**